bot/util/utilities.py

```diff
@@ -86,6 +90,72 @@ async def about(self, ctx):
         embed.add_field(name="Links:", value=str_links)
         await ctx.send(embed=embed)
 
+    @commands.command(name='embed')
+    async def embed(self, ctx, channel: str, color: str, *, text: str):
+        """Command Handler for the embed command
+
+        Creates and sends an embed in the specified channel with color and title and text. Title and text are separated
+        by a '|' character.
+
+        Args:
+            ctx (Context): The context in which the command was called.
+            channel (str): The channel where to post the message. Can be channel name (starting with #) or channel id.
+            color (str): Color code for the color of the strip.
+            text (str): The text to be posted in the embed. The string contains title and content, which are separated
+                        by a '|' character. If this character is not found, no title will be assumed.
+        """
+        try:
+            channel_to_post = get_text_channel(channel, ctx.guild)
+        except ValueError as error:
+            await ctx.send(error)
+            return
+
+        if '|' in text:
+            title, description = text.split('|')
+        else:
+            title = ''
+            description = text
+
+        try:
+            col = discord.Color(int('0x' + color, 16))
+        except ValueError:
+            await ctx.send("Color could not be parsed. Please use a valid color code.")
+            return
+
+        embed = discord.Embed(title=title, description=description, color=col)
+        await channel_to_post.send(embed=embed)
+
+    @commands.command(name='cembed')
+    async def cembed(self, ctx, channel: str, *, json_string: str):
+        """Command Handler for the embed command.
+
+        Creates and sends an embed in the specified channel parsed from json.
+
+        Args:
+            ctx (Context): The context in which the command was called.
+            channel (str): The channel where to post the message. Can be channel name (starting with #) or channel id.
+            json (str): The json string representing the embed. Alternatively it could also be a pastebin link
+        """
+        try:
+            channel_to_post = get_text_channel(channel, ctx.guild)
+            if is_pastebin_link(json_string):
+                json_string = parse_pastebin_link(json_string)
+            embed_dict = json.loads(json_string)
+            embed = discord.Embed.from_dict(embed_dict)
+            if len(embed) == 0:
```

Comment:
This check doesn't work. If I remember correctly an Embed needs to have at least a title and a description if it doesn't have any additional fields. If it has additional fields at least they need to have a title and some text.
You can try to verify this with your code.

Reply:
Verified by doc that all fields on an embed are optional but if it has 0 valid fields then it will break when trying to send it.
The docs state that len is implemented and is the number of fields that are filled (validly)

Reply:
I just tested this with carlbot and an embed breaks if an additional field doesn't have a title even though there are other valid fields in the embed. Don't know if this is Carls fault though.

Reply:
Oh sry, thought you meant the embed itself needs a title. Will check the doc in additional fields in embeds and try to find graceful behavior in case of this